connectome_interpreter/external_map.py

```diff
@@ -150,6 +161,154 @@ def map_to_experiment(df, dataset=None, custom_experiment=None):
     )
     return target2chem
 
+def hex_heatmap(
+    df:pd.Series
+    , style:dict=None
+    , sizing:dict=None
+) -> go.Figure:
+    """
+    Generate a hexagonal heat map plot of the data in a pandas series 'df'.
+
+    Args: 
+        df : pd.Series
+            A Series where the index is formatted as 'x,y' coordinates and values represent data to plot.
+        style : dict, default=None
+            Dict containing styling formatting variables.
+        sizing : dict, default=None
+            Dict containing size formatting variables.
+
+    Returns:
+        fig : go.Figure
+    """
+    # Default styling and sizing parameters to use if not specified.
+    if style is None:
+        style = {
+            "font_type": "arial",
+            "markerlinecolor": "rgba(0,0,0,0)", #transparent
+            "linecolor": "black",
+            "papercolor": "rgba(255,255,255,255)"
+        }
+
+    if sizing is None:
+        sizing = {
+            "fig_width": 260,  # units = mm
+            "fig_height": 220,  # units = mm
+            "fig_margin": 0,
+            "fsize_ticks_pt": 20,
+            "fsize_title_pt": 20,
+            "markersize": 18,
+            "ticklen": 15,
+            "tickwidth": 5,
+            "axislinewidth": 3,
+            "markerlinewidth": 0.9,
+            "cbar_thickness": 20,
+            "cbar_len": 0.75,
+        }
+
+    # sizing of the figure and font
+    pixelsperinch = 72 # for svg and pdf
+    pixelspermm = pixelsperinch / 25.4
+    area_width = (sizing["fig_width"] - sizing["fig_margin"]) * pixelspermm
+    area_height = (sizing["fig_height"] - sizing["fig_margin"]) * pixelspermm
+    fsize_ticks_px = sizing["fsize_ticks_pt"] * (1 / 72) * pixelsperinch
+    fsize_title_px = sizing["fsize_title_pt"] * (1 / 72) * pixelsperinch
+
+    # Convert index values (formatted as '-12,34') into separate x and y coordinates
+    coords = [tuple(map(int, idx.split(","))) for idx in df.index]
+    x_vals, y_vals = zip(*coords)  # Separate into x and y lists
+
+    # initiate plot
+    fig = go.Figure()
+    fig.update_layout(
+        autosize=False
+      , height=area_height
+      , width=area_width
+      , margin={"l": 0, "r": 0, "b": 0, "t": 0, "pad": 0}
+      , paper_bgcolor=style["papercolor"]
+      , plot_bgcolor=style["papercolor"]
+    )
+    fig.update_xaxes(
+        showgrid=False, showticklabels=False, showline=False, visible=False
+    )
+    fig.update_yaxes(
+        showgrid=False, showticklabels=False, showline=False, visible=False
+    )
+    # Symbol number to choose to plot hexagons
+    symbol_number = 15
+
+    # Get the coordinates of all columns in the medulla:
+    col_coords = pd.read_csv("../connectome_interpreter/data/Nern2024/ME-column-coords.csv")
```

Comment:
This hex heatmap plotter appears to be general, but this line suggests it's specific for plotting one particular dataset?

Reply:
Thank you. It is indeed so far for one particular dataset: there are two datasets (eyes of flies) in total, but the number/position of the units in the eye are slightly different between the individuals. So for now we'll just use one. Will incorporate the other at some point.